### PR TITLE
feat(ecs): machineImageType support AL2023

### LINKED ISSUE #32496: (ecs): Support AL2023
### Describe the feature

ECS support AL2023

### Use Case

ECS support AL2023

### Proposed Solution

ECS [MachineImageType](https://github.com/aws/aws-cdk/blob/9d5ad934caf2c5b30967535631da2caba3b62f0a/packages/aws-cdk-lib/aws-ecs/lib/cluster.ts#L93) add `AMAZON_LINUX_2023`

### Other Information

As instruction https://github.com/aws/aws-cdk/issues/32469#issuecomment-2536820262

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

latest

### Environment details (OS name and version, etc.)

Amazon Linux 2023

### PR DESCRIPTION
**Pull request by phuhung273**
### Issue # (if applicable)

Closes #32496
Relate #32469 

### Reason for this change

- Invalid user data on AL2023

### Description of changes

- ECS `machineImageType` support AL2023

### Description of how you validated changes

Unit + Integration test

#### Instance can join cluster
image
#### iptables command success
image

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-ecs/lib/cluster.ts [hunk 1]
```diff
@@ -631,6 +622,31 @@ export class Cluster extends Resource implements ICluster {
...
+    autoScalingGroup.addUserData(`echo ECS_CLUSTER=${this.clusterName} >> /etc/ecs/ecs.config`);
+    if (!options.canContainersAccessInstanceRole) {
+      // Deny containers access to instance metadata service
+      // Source: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/instance_IAM_role.html
+      autoScalingGroup.addUserData('sudo iptables --insert FORWARD 1 --in-interface docker+ --destination 169.254.169.254/32 --jump DROP');
+
+      switch (options.machineImageType) {
+        case MachineImageType.AMAZON_LINUX_2023: {
+          autoScalingGroup.addUserData('sudo iptables-save');
+          break;
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-ecs/lib/cluster.ts:635]: Can you confirm for `AL2023`, the old command `sudo service iptables save` doesn't work to make sure this is not a breaking change?

**Review by GavinZZ — CHANGES_REQUESTED** [review]
Just a comment to confirm prior to approving

File: packages/aws-cdk-lib/aws-ecs/lib/cluster.ts (hunk 1 above)
Comment (phuhung273) [reply packages/aws-cdk-lib/aws-ecs/lib/cluster.ts:635]: image

Thanks for your review @GavinZZ. This is the old command on latest ECS optimized AL2023 `ami-0a357ea20d7b79c2c`

**Review by GavinZZ — APPROVED** [review]

**Comment by GavinZZ** [comment]
@phuhung273 sorry that I had to revert this PR. As I just discussed with my team internally, there's known issue with the flag usage `canContainersAccessInstanceRole` and we're deprecating this flag (hopefully next Wednesday's release), so we should not introducing any new feature to this property that will result in more users using this feature. Apologize for the inconvenience.

**Pull request merged**